### PR TITLE
New vocabulary DataIntegration

### PR DESCRIPTION
**Pull request by biedenstein**
Hi,

I created the DataIntegration vocabulary. I executed `npm run build` on my local machine to generate DataIntegraion.json and DataIntegration.md.

Best regards,
Stefan

### CODE REVIEW

File: vocabularies/DataIntegration.xml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    <edmx:Include Alias="Core" Namespace="Org.OData.Core.V1" />
+  </edmx:Reference>
+
+  <edmx:DataServices>
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="com.sap.vocabularies.DataIntegration.v1" Alias="DataIntegration">
+      <Annotation Term="Core.Description">
+        <String>Terms for annotating analytical resources</String>
+      </Annotation>
+      <Annotation Term="Core.Description" Qualifier="Published">
+        <String>2017-02-15 © Copyright 2013 SAP AG. All rights reserved</String>
```
Comment (ralfhandl) [inline vocabularies/DataIntegration.xml:13]:
```suggestion
        <String>2021-02-11 © Copyright 2021 SAP AG. All rights reserved</String>
```

File: vocabularies/DataIntegration.xml [hunk 2]
```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+  <edmx:Reference Uri="https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Core.V1.xml">
+    <edmx:Include Alias="Core" Namespace="Org.OData.Core.V1" />
+  </edmx:Reference>
+
+  <edmx:DataServices>
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="com.sap.vocabularies.DataIntegration.v1" Alias="DataIntegration">
+      <Annotation Term="Core.Description">
+        <String>Terms for annotating analytical resources</String>
```
Comment (ralfhandl) [inline vocabularies/DataIntegration.xml:10]:
```suggestion
        <String>Terms for Data Integration</String>
```
Maybe this title?

File: vocabularies/DataIntegration.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        <Annotation Term="Core.Example">
+          <Record>
+            <Annotation Term="com.sap.vocabularies.DataIntegration.v1.OriginalDataType">
+							<Collection>
+								<Record>
+									<PropertyValue Property="Property" Path="CostCenterID"/>
+									<PropertyValue Property="OriginalDataType" String="CHAR(000010)"/>
+								</Record>
+							</Collection>
+						</Annotation>
```
Comment (ralfhandl) [inline vocabularies/DataIntegration.xml:47]:
```suggestion
            <Annotation Term="com.sap.vocabularies.DataIntegration.v1.OriginalDataType" String="CHAR(000010)" />
```

The type of the term is `Edm.String`.

File: vocabularies/DataIntegration.xml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        <Annotation Term="Core.Example">
+          <Record>
+            <Annotation Term="com.sap.vocabularies.DataIntegration.v1.ConversionExit">
+							<Collection>
+								<Record>
+									<PropertyValue Property="Property" Path="CostCenterID"/>
+									<PropertyValue Property="ConversionExit" String="ALPHA"/>
+								</Record>
+							</Collection>
+						</Annotation>
```
Comment (ralfhandl) [inline vocabularies/DataIntegration.xml:63]:
```suggestion
            <Annotation Term="com.sap.vocabularies.DataIntegration.v1.ConversionExit" String="ALPHA" />
```

Term type is `Edm.String`

File: vocabularies/DataIntegration.xml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+				</Member>
+				<Member Name="UPDATE">
+					<Annotation Term="Core.Description" String="Delta is supported for updates"/>
+				</Member>
+				<Member Name="DELETE">
+					<Annotation Term="Core.Description" String="Delta is supported for deletes"/>
+				</Member>
+			</EnumType>
+			<Term Name="DeltaMethod" Type="DataIntegration.DeltaMethodType" AppliesTo="EntitySet">
+				<Annotation Term="Core.Description" String="Defines which delta method the entity set supports. Only evaluated if Capabilities.ChangeTracking Supported is true"/>
```
Comment (ralfhandl) [inline vocabularies/DataIntegration.xml:83]:
```suggestion
				<Annotation Term="Core.Description" String="Defines which delta method the entity set supports. Only evaluated if Capabilities.ChangeTracking/Supported is true"/>
```

File: vocabularies/DataIntegration.xml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+						</Annotation>
+          </Record>
+				</Annotation>
+			</Term>
+			<Term Name="SourceSystem" Type="Edm.String" AppliesTo="Container">
+				<Annotation Term="Core.Description" String="Identifier that classifies the type of the source system"/>
+        <Annotation Term="Core.LongDescription" String="The original type name used in annotation OriginalDataType depend are specific to different source system. Sourc system type ABAP uses other type names as source system type HANA." />
+			</Term>
+			<EnumType Name="DeltaMethodType" IsFlags="true">
+				<Member Name="INSERT">
```
Comment (ralfhandl) [inline vocabularies/DataIntegration.xml:72]:
```suggestion
				<Member Name="INSERT" Value="1">
```

Flags enumeration members must have a numeric value

File: vocabularies/DataIntegration.xml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+					<Annotation Term="Core.Description" String="Delta is supported for inserts"/>
+				</Member>
+				<Member Name="UPDATE">
```
Comment (ralfhandl) [inline vocabularies/DataIntegration.xml:75]:
```suggestion
				<Member Name="UPDATE" Value="2">
```

File: vocabularies/DataIntegration.xml [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+					<Annotation Term="Core.Description" String="Delta is supported for updates"/>
+				</Member>
+				<Member Name="DELETE">
```
Comment (ralfhandl) [inline vocabularies/DataIntegration.xml:78]:
```suggestion
				<Member Name="DELETE" Value="4">
```

**Review by ralfhandl — CHANGES_REQUESTED** [review]
Looks good overall, see comments for minor required changes

**Comment by biedenstein** [comment]
Hi Ralf,

It hit somewhere an auto-refresh-button. Now there are several .md files part of this PR and if have no idea how to resolve it.

BR, Stefan

**Review by ralfhandl — DISMISSED** [review]

**Comment by ralfhandl** [comment]
Hi Stefan,

Is this vocabulary already used by a shipped implementation?

If not, I suggest marking it as "experimental".

Thanks in advance
Ralf

**Comment by biedenstein** [comment]
Hi Ralf,

Yes, the vocabulary is already used in SAP Marketing Cloud and SAP S/4HANA. [SAP help]((https://help.sap.com/viewer/0f9408e4921e4ba3bb4a7a1f75f837a7/2102.500/en-US/b4692a509a9a4e20a81e08fb8749b25d.html)) Additionally, there are consumers in SAP HANA SDI and SAP Data Intelligence.

Best regards,
Stefan

**Review by ralfhandl — APPROVED** [review]

**Pull request merged**